src/main/java/ConsoleOut/ConsoleOut.java

```diff
@@ -1,19 +1,45 @@
 package ConsoleOut;
+
 import Main.History;
+import Main.Way;
+
+import java.util.Map;
+import java.util.Set;
 
 public class ConsoleOut implements UserWriter {
-    private TextWriter textWriter;
-    private WaysWriter waysWriter;
     private int delay;
     private String color;
+
     public ConsoleOut(int delay, String color) {
-        this.textWriter = new TextWriter();
-        this.waysWriter = new WaysWriter();
-        this.color= ColorCoder.getColorCode(color);
+        this.color = ColorCoder.getColorCode(color);
         this.delay = delay;
     }
-    public void writeHistory(History history){
-        textWriter.Write(history.getHistory(), delay, color);
-        waysWriter.Write(history.getWays(), delay, color);
+
+    public void writeHistory(History history) {
+        writeStory(history.getHistory());
+        writeWays(history.getWays());
+    }
+
+    void writeStory(String text) {
+        for (char c : text.toCharArray()) {
```

Comment:
If You want, try to learn something about streams in Java (collection streams) and replace this for each loop with a stream, or a for each with lamba ;)